### PR TITLE
BUGFIX: CL-10460 Modified format of metric is not applied on filtered metric and PoP metric

### PR DESCRIPTION
**Pull request by akloboucnik**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded.

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/233/ : SUCCESS in 2m 50s

File: src/execution.js [hunk 1]
```diff
@@ -477,7 +478,28 @@ export const mdToExecutionConfiguration = (mdObj, options = {}) => {
     };
 };
 
+const getOriginalMetricFormats = (mdObj) => {
+    const measures = map(get(mdObj, 'buckets.measures'), ({ measure }) => measure);
+    // for metrics with showPoP or measureFilters.length > 0 roundtrip for original metric format
+    return $.when.apply(this, map(measures, (measure) => {
+        if (measure.showPoP === true || measure.measureFilters.length > 0) {
+            return xhrGet(measure.objectUri).then((obj) => {
+                measure.format = get(obj, 'metric.content.format', measure.format);
```
Comment (vaclavbohac) [inline src/execution.js]: I wouldn't mutate the measure:

```js
return {
    ...measure,
    format: get(/* ... */)
};
```

File: src/execution.js [hunk 2]
```diff
@@ -477,7 +478,28 @@ export const mdToExecutionConfiguration = (mdObj, options = {}) => {
     };
 };
 
+const getOriginalMetricFormats = (mdObj) => {
+    const measures = map(get(mdObj, 'buckets.measures'), ({ measure }) => measure);
+    // for metrics with showPoP or measureFilters.length > 0 roundtrip for original metric format
+    return $.when.apply(this, map(measures, (measure) => {
```
Comment (vaclavbohac) [inline src/execution.js]: I don't know if it's clear what `this` is in the top level arrow function.

File: src/execution.js [hunk 3]
```diff
@@ -477,7 +478,28 @@ export const mdToExecutionConfiguration = (mdObj, options = {}) => {
...
+        return $.Deferred().resolve(measure);
+        /* eslint-enable new-cap */
+    }));
+};
+
 export const getDataForVis = (projectId, mdObj, settings) => {
-    const { columns, ...executionConfiguration } = mdToExecutionConfiguration(mdObj);
-    return getData(projectId, columns, executionConfiguration, settings);
+    return getOriginalMetricFormats(mdObj).then((...args) => {
+        const measures = map([...args], (measure) => { return { measure: measure }; });
```
Comment (vaclavbohac) [inline src/execution.js]:
```js
(measure) => ({ measure })
```

File: src/execution.js [hunk 4]
```diff
@@ -477,7 +478,28 @@ export const mdToExecutionConfiguration = (mdObj, options = {}) => {
...
+        /* eslint-disable new-cap */
+        return $.Deferred().resolve(measure);
+        /* eslint-enable new-cap */
+    }));
+};
+
 export const getDataForVis = (projectId, mdObj, settings) => {
-    const { columns, ...executionConfiguration } = mdToExecutionConfiguration(mdObj);
-    return getData(projectId, columns, executionConfiguration, settings);
+    return getOriginalMetricFormats(mdObj).then((...args) => {
```
Comment (vaclavbohac) [inline src/execution.js]: `...measures` instead of `...args` would be more readable

Comment (vaclavbohac) [inline src/execution.js]: You can avoid eslint error by inlining the map function

File: test/execution_test.js [hunk 5]
```diff
@@ -971,5 +971,141 @@ describe('execution', () => {
...
+                                'type': 'date',
+                                'collection': 'attribute',
+                                'displayForm': '/gdc/md/myFakeProjectId/obj/1234',
+                                'attribute': '/gdc/md/myFakeProjectId/obj/1233'
+                            }
+                        }
+                    ]
+                }
+            };
+            beforeEach(() => {
```
Comment (vaclavbohac) [inline test/execution_test.js]: I would add one space above `beforeEach` O:-)

**Comment by gdgate** [comment]
Build succeeded.

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/234/ : SUCCESS in 2m 53s

**Pull request merged**